torch/csrc/jit/script/compiler.cpp

```diff
@@ -817,12 +820,15 @@ struct to_ir {
     // it is not a real thing yet, so just say the type is None
     closure_node->output()->setType(NoneType::get());
     Block* block = closure_node->addBlock();
+    auto prev_status = loop_status_;
+    loop_status_ = LoopStatus::NOT_IN_LOOP;
```

Comment:
Let's make this a resource guard to make it consistent with the WithInsertPoint below.